packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts

```diff
@@ -257,19 +261,51 @@ export const getDefaultCountry = (country: string = regional.unknownCountry) =>
     };
 };
 
-// fill ids and remove alternative quotes
-export function processSellAndBuyQuotes<T extends CoinmarketTradeBuySellType>(
-    allQuotes: CoinmarketTradeBuySellDetailMapProps[T][] | undefined,
-): CoinmarketTradeBuySellDetailMapProps[T][] {
+export const filterQuotesAccordingTags = <T extends CoinmarketTradeBuySellType>(
+    allQuotes: CoinmarketTradeBuySellDetailMapProps[T][],
+) => {
+    return allQuotes.filter(q => !q.tags || !q.tags.includes('alternativeCurrency'));
+};
+
+// fill orderId for all and paymentId for sell and buy
+export const addIdsToQuotes = <T extends CoinmarketTradeType>(
+    allQuotes: CoinmarketTradeDetailMapProps[T][] | undefined,
+    type: CoinmarketTradeType,
+): CoinmarketTradeDetailMapProps[T][] => {
     if (!allQuotes) allQuotes = [];
+
     allQuotes.forEach(q => {
-        q.orderId = uuidv4();
+        const sellBuyQuote = ['buy', 'sell'].includes(type)
+            ? (q as BuyTrade | SellFiatTrade)
+            : null;
 
-        if (!q.paymentId) {
-            q.paymentId = uuidv4();
+        if (sellBuyQuote && !sellBuyQuote.paymentId) {
+            sellBuyQuote.paymentId = uuidv4();
         }
+
+        q.orderId = uuidv4();
     });
-    const quotes = allQuotes.filter(q => !q.tags || !q.tags.includes('alternativeCurrency'));
 
-    return quotes;
-}
+    return allQuotes;
+};
+
+export const getBestRatedQuote = (
+    quotes: CoinmarketTradeDetailType[] | undefined,
+    type: CoinmarketTradeType,
+): CoinmarketTradeDetailType | undefined => {
+    const quotesFiltered = quotes?.filter(item => item.rate && item.rate !== 0);
+    const bestRatedQuotes = quotesFiltered
+        ? [...quotesFiltered].sort((a, b) => {
+              // ascending to rate for buy - lower rate more crypto client receives
+              if (type === 'buy') {
+                  return new BigNumber(a.rate ?? 0).minus(new BigNumber(b.rate ?? 0)).toNumber();
+              }
+
+              // descending to rate for sell/exchange - higher rate more crypto/fiat client receives
+              return new BigNumber(b.rate ?? 0).minus(new BigNumber(a.rate ?? 0)).toNumber();
+          })
+        : null;
+    const bestRatedQuote = bestRatedQuotes?.[0];
+
+    return bestRatedQuote;
+};
```

Comment:
Maybe move the logic to server? cc: @FreeWall

Reply:
Best rated quote (or `quotes` array in this case) depends on client-side filtering (payment method), so it's calculated on client-side.
In order to use logic on server side, we would need to calculate best rate for each payment method and pass the info to client somehow.